src/libraries/Microsoft.PowerFx.Core/Public/Types/TableType.cs

```diff
@@ -31,10 +32,20 @@ public override void Visit(ITypeVistor vistor)
             vistor.Visit(this);
         }
 
-        public TableType Add(NamedFormulaType field)
+        public TableType Add(NamedFormulaType field, string optionalDisplayName = null)
         {
+            var displayNameProvider = _displayNameProvider;
+            if (optionalDisplayName != null)
+            {
+                if (displayNameProvider == null)
+                    displayNameProvider = new DisplayNameProvider();
+
+                if (!displayNameProvider.TryAddField(field.Name, optionalDisplayName))
+                    throw new NameCollisionException(optionalDisplayName);
```

Comment:
It would be more defensive to have this be AddField() and have the method throw. 

IE, when would somebody ever call TryAddField and *not* throw? That could open the door to some non-deterministic behavior (behavior depending on the order of calls to Try).

Reply:
yup, updated, and made more immutable.